File: tests/fixtures/c/comprehensive.c
// Comprehensive C fixture -- covers all 5 DefKind: Function, Struct, Enum, Type, Const

// === Function ===

int process(int x) {
    return x * 2;
}

static void helper(void) {
    // static function
}

char *dup_str(const char *s) {
    return ((void*)0);
}

// === Struct ===

struct Point {
    double x;
    double y;
};

struct Config {
    int timeout;
    int retries;
};

// === Enum ===

enum Color {
    RED,
    GREEN,
    BLUE
};

enum Status {
    OK = 0,
    ERROR = 1
};

// === Type (typedef) ===

typedef int StatusCode;

typedef char *StringPtr;

typedef struct Node Node;

// === Const ===

const int MAX_SIZE = 1024;

static const int VERSION = 2;

const char *MSG = "hello";

// === Definitions inside function body (should NOT be extracted) ===

void withLocalDefs(void) {
    const int LOCAL_CONST = 42;
}

File: tests/fixtures/dart/comprehensive.dart
// Dart comprehensive test fixture for peek
// Covers: class, abstract class, enum, mixin, extension, typedef, const, function,
//         getter/setter, constructor, factory constructor, nested scope, annotations

// Top-level function
void globalHelper(String message) {
  print(message);
}

// Top-level constants
const APP_VERSION = "1.0.0";
const int MAX_RETRIES = 3;

// Type alias
typedef Callback = void Function(String);

// Simple class with annotation
@deprecated
class UserService {
  final String name;

  // Constructor
  UserService(this.name);

  // Factory constructor
  factory UserService.admin() => UserService("admin");

  // Method
  String greet(String greeting) {
    return "$greeting, $name!";
  }

  // Getter
  String get displayName => name.toUpperCase();

  // Setter
  set displayName(String value) {
    // setter logic
  }

  // Static const
  static const int DEFAULT_TIMEOUT = 30;
}

// Abstract class
abstract class BaseProcessor {
  void process();

  String describe() => "Processor";
}

// Enum
enum Status { active, inactive, pending }

// Mixin
mixin Loggable {
  void log(String message) {
    print(message);
  }
}

mixin Validatable {
  bool validate() => true;
}

// Extension
extension StringExt on String {
  String repeated(int times) => this * times;

  String get capitalized =>
      isEmpty ? '' : '${this[0].toUpperCase()}${substring(1)}';
}

// Extension type
extension type Point(double x, double y) {
  Point.origin() : this(0.0, 0.0);

  double distanceTo(Point other) {
    return (x - other.x).abs() + (y - other.y).abs();
  }
}

// Class with mixin
class ApiClient extends BaseProcessor with Loggable {
  @override
  void process() {
    log("Processing...");
  }
}

// Multi const declaration
const CACHE_TTL = 300, MAX_ITEMS = 1000;

// Definitions inside function body (should NOT be extracted)
void withLocalDefs() {
  const localConst = 42;
  void localHelper() {
    print("local");
  }
}

// Class with various fields
class Product {
  int id;
  String label;
  static int instanceCount = 0;
  final double price;

  Product(this.id, this.label, this.price);
}
